Util/Inflector/DoctrineInflector.php

```diff
@@ -17,9 +17,16 @@
  * Inflector object using the Doctrine/Inflector.
  *
  * @author Mark Kazemier <Markkaz>
+ *
+ * @deprecated since 1.7, to be removed in 2.0. Use {@link \FOS\RestBundle\Inflector\DoctrineInflector} instead.
  */
 class DoctrineInflector implements InflectorInterface
 {
+    public function __construct()
+    {
+        @trigger_error('FOS\RestBundle\Util\Inflector\DoctrineInflector is deprecated since version 1.7 and will be removed in 2.0. Use FOS\RestBundle\Inflector\DoctrineInflector instead.', E_USER_DEPRECATED);
```

Comment:
instead of overwriting the constructor in the child class to avoid the warning, you can use ``if (!$this instanceof NewDoctrineInflector) { @trigger_error(...) }`` (if it is an instance of the new class, it means the code is using the new class and so things are OK)